### DISCUSSION TITLE
How to create a mapping for telescope's grep_string?

### DISCUSSION THREAD

**Discussion by nkeil** [NvChad/NvChad#2130] [Q&A] [answered]
I've been attempting to mirror this code in the NvChad `mappings.lua` config:

```lua
local builtin = require('telescope.builtin')

vim.keymap.set('n', '‹leader>ps', function()
    builtin.grep_string({ search = vim.fn.input("Grep > ") });
end
```

But I seem to be running into issues. I get `module 'telescope.builtin' not found`, which I think has to do with lazy loading? I was able to get some other maps to work by using things like `":Telescope find_files <CR>"`, but I'm not sure how to pass the input argument into a command this way. This is what my code in `mappings.lua` looks like:

```lua
local builtin = require('telescope.builtin')
-- ...
M.telescope = {
  n = {
    ["<leader>pf"] = { ":Telescope find_files <CR>", "Find project files" },
    ["<C-p>"] = { ":Telescope git_files <CR>", "Find git files" },
    ["<leader>ps"] = {
      function()
        telescope.builtin.grep_string({ seach = vim.fn.input("Grep > ") });
      end,
      "Search project files"
    },
  }
}
return M
```

**Comment by siduck** [comment, marked as answer]
require("telescope.builtin").grep_string...

**Reply by nkeil** [reply]
Thank you! Putting the require statement inside the function worked
